### PR TITLE
Clean up makefiles

### PR DESCRIPTION
**Pull request by andressbarajas**
Clean up example makefiles

### CODE REVIEW

**Comment by gyrovorbis** [comment]
Good work! I was just noticing how inconsistent they were after adding the last example. lol.

**Comment by QuzarDC** [comment]
So, perhaps a silly question, but ... is there a good reason we don't add a lot of these pieces into Makefile.rules? That way there's no worry about having 100 copies that are all the same.

**Comment by andressbarajas** [comment]
I dont see any pieces that are 100% shared across all examples.

**Comment by ljsebald** [comment]
I'd really rather that we didn't remove the `-` characters before the `rm` commands. That is there by design, and should stay.

**Comment by andressbarajas** [comment]
I reverted the removal of the dashes and added them to rm in dist:

**Comment by andressbarajas** [comment]
Also as a note:

Using -rm in a Makefile: When you use -rm instead of rm in a Makefile, any errors encountered during the removal process are ignored, and the make process continues even if the removal fails. This allows the build process to proceed without interruptions caused by non-critical removal errors.

File: examples/dreamcast/2ndmix/Makefile [hunk 1]
```diff
@@ -13,10 +13,10 @@ KOS_ROMDISK_DIR = romdisk_boot
 include $(KOS_BASE)/Makefile.rules
 
 clean: rm-elf
-	rm -f $(OBJS)
+	-rm -f $(OBJS)
```
Comment (jpeach) [inline examples/dreamcast/2ndmix/Makefile:16]: Not really an issue, but if you like you could use the `$(RM)` variable here (it expands to `rm -f`).

https://www.gnu.org/software/make/manual/html_node/Implicit-Variables.html#index-RM

Comment (andressbarajas) [reply examples/dreamcast/2ndmix/Makefile:16]: I think we want to keep -rm instead of rm.

**Review by ljsebald — APPROVED** [review]

**Pull request merged**